### PR TITLE
Add support for reading/writing TMX file data

### PR DESCRIPTION
**Pull request by RezMike**
Almost full support for reading and writing data from/to TMX files.
Not yet supported features:
- zstd image compression,
- object templates.

### CODE REVIEW

File: src/main/kotlin/com/soywiz/korge/intellij/editor/tiled/TiledMapWriter.kt [hunk 1]
```diff
@@ -223,13 +315,13 @@ private fun XmlBuilder.propertiesToXml(properties: Map<String, TiledMap.Property
...
 		properties.forEach { (name, prop) ->
 			when (prop) {
-				is TiledMap.Property.StringT -> property(name, "string", prop.value)
-				is TiledMap.Property.IntT -> property(name, "int", prop.value)
-				is TiledMap.Property.FloatT -> property(name, "float", prop.value)
-				is TiledMap.Property.BoolT -> property(name, "bool", prop.value.toString())
-				is TiledMap.Property.ColorT -> property(name, "color", prop.value.toStringARGB())
-				is TiledMap.Property.FileT -> property(name, "file", prop.path)
-				is TiledMap.Property.ObjectT -> property(name, "object", prop.id)
+				is Property.StringT -> property(name, "string", prop.value)
```
Comment (soywiz) [inline src/main/kotlin/com/soywiz/korge/intellij/editor/tiled/TiledMapWriter.kt:420]: Even when in packages I usually like to either limit the scope of the class by putting in inside other class like TiledMap.Property, or to prefix with TiledMapProperty. The reason is that with global autocompletion there might be several unrelated Property classes that might be confusing to people or annoying when importing the wrong one

File: src/main/kotlin/com/soywiz/korge/intellij/editor/tiled/TiledMap.kt [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+	class WangTile(
+		val tileId: Int,
+		val wangId: Int,
+		val hflip: Boolean = false,
+		val vflip: Boolean = false,
+		val dflip: Boolean = false
+	)
+}
+
+class EditorSettings(
```
Comment (soywiz) [inline src/main/kotlin/com/soywiz/korge/intellij/editor/tiled/TiledMap.kt:96]: Same here. Can you rename it to `TiledEditorSettings`?

File: src/main/kotlin/com/soywiz/korge/intellij/editor/tiled/TiledMapWriter.kt (hunk 1 above)
Comment (RezMike) [reply src/main/kotlin/com/soywiz/korge/intellij/editor/tiled/TiledMapWriter.kt:420]: Well, here the import for `TiledMap.*` was added, and IDEA suggested remove redundant notices

Comment (RezMike) [reply src/main/kotlin/com/soywiz/korge/intellij/editor/tiled/TiledMapWriter.kt:420]: then it's okay, good

File: src/main/kotlin/com/soywiz/korge/intellij/editor/tiled/TiledMap.kt (hunk 2 above)
Comment (RezMike) [reply src/main/kotlin/com/soywiz/korge/intellij/editor/tiled/TiledMap.kt:96]: I think I will probably move this class inside TiledMap

**Comment by soywiz** [comment]
Screenshot from 2020-07-05 14-52-31
Screenshot from 2020-07-05 14-52-34

**Comment by RezMike** [comment]
Drawing is not implemented yet. In this PR I just added the ability to read and write tmx file, not to draw its content.
About this exception - I know about it and I'm fixing it right now. I will also add some format fixes (order of attributes, default values, etc.)

**Comment by RezMike** [comment]
@soywiz so what are your thoughts about this PR after I added the last commit? Can it be merged or there are some other changes I should add here?

**Comment by soywiz** [comment]
Let me try to draw something with the plugin

**Pull request merged**